src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SuddenHighwayTypeChangeCheckTest.java

```diff
@@ -0,0 +1,39 @@
+package org.openstreetmap.atlas.checks.validation.linear.edges;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.openstreetmap.atlas.checks.configuration.ConfigurationResolver;
+import org.openstreetmap.atlas.checks.validation.verifier.ConsumerBasedExpectedCheckVerifier;
+
+/**
+ * Tests for {@link SuddenHighwayTypeChangeCheck}
+ *
+ * @author v-garei
+ */
+public class SuddenHighwayTypeChangeCheckTest
+{
+
+    @Rule
+    public SuddenHighwayTypeChangeCheckTestRule setup = new SuddenHighwayTypeChangeCheckTestRule();
+
+    @Rule
+    public ConsumerBasedExpectedCheckVerifier verifier = new ConsumerBasedExpectedCheckVerifier();
+
+    private final SuddenHighwayTypeChangeCheck check = new SuddenHighwayTypeChangeCheck(
+            ConfigurationResolver.inlineConfiguration(
+                    "{\"SuddenHighwayTypeChangeCheck\": {\"minHighwayType\": \"tertiary\"}}"));
+
+    @Test
+    public void testFalsePositiveSuddenHighwayTypeChangeCheck()
+    {
+        this.verifier.actual(this.setup.falsePositiveSuddenHighwayTypeChangeCheck(), this.check);
+        this.verifier.verifyEmpty();
+    }
+
+    @Test
+    public void testTruePositiveSuddenHighwayTypeChangeCheck()
```

Comment:
It would be good to have a true positive tests for each of the 3 cases.

Reply:
each case has been added in test